src/main/java/com/razorpay/OAuthTokenClient.java

```diff
@@ -0,0 +1,126 @@
+                + "&client_id=" + clientId
+                + "&redirect_uri=" + redirectUri
+                + scopesArray
+                + "&state=" + state;
+
+        return AuthorizeUrl;
+    }
+
+    public OauthToken getAccessToken(JSONObject request) throws RazorpayException {
+        validateAccessTokenRequest(request);
+        return post(null, Constants.TOKEN, request, Constants.AUTH);
+    }
+
+    public OauthToken refreshToken(JSONObject request) throws RazorpayException {
+        validateRefreshTokenRequest(request);
+        request.put(GRANT_TYPE, REFRESH_TOKEN);
+        return post(null, Constants.TOKEN, request, Constants.AUTH);
+    }
+
+    public OauthToken revokeToken(JSONObject request) throws RazorpayException {
+        validateRevokeTokenRequest(request);
+        return post(null, Constants.REVOKE, request, Constants.AUTH);
+    }
+
+    private String[] jsonArrayToStringArray(JSONArray jsonArray) {
+        String[] array = new String[jsonArray.length()];
+        for (int i = 0; i < jsonArray.length(); i++) {
+            array[i] = jsonArray.getString(i);
+        }
+        return array;
+    }
+
+    private void validateAuthURLRequest(JSONObject request) throws RazorpayException {
+        payloadValidator.validate(request, getValidationsForAuthRequestURL());
+    }
+
+    private void validateAccessTokenRequest(JSONObject request) throws RazorpayException {
+        payloadValidator.validate(request, getValidationsForAccessTokenRequest());
+    }
+
+    private void validateRefreshTokenRequest(JSONObject request) throws RazorpayException {
+        payloadValidator.validate(request, getValidationsForRefreshTokenRequest());
+    }
+
+    private void validateRevokeTokenRequest(JSONObject request) throws RazorpayException {
+        payloadValidator.validate(request, getValidationsForRevokeTokenRequest());
+    }
+
+    private List<ValidationConfig> getValidationsForAuthRequestURL() {
+        return Arrays.asList(
+                new ValidationConfig(CLIENT_ID, Collections.singletonList(ValidationType.ID)),
+                new ValidationConfig(REDIRECT_URI, Arrays.asList(ValidationType.NON_EMPTY_STRING, ValidationType.URL)),
+                new ValidationConfig(SCOPES, Collections.singletonList(ValidationType.NON_NULL)),
+                new ValidationConfig(STATE, Collections.singletonList(ValidationType.NON_EMPTY_STRING))
+        );
+    }
+
+    private List<ValidationConfig> getValidationsForAccessTokenRequest() {
+        return Arrays.asList(
+                new ValidationConfig(CLIENT_ID, Collections.singletonList(ValidationType.ID)),
+                new ValidationConfig(CLIENT_SECRET, Collections.singletonList(ValidationType.NON_EMPTY_STRING)),
+                new ValidationConfig(REDIRECT_URI, Arrays.asList(ValidationType.NON_EMPTY_STRING, ValidationType.URL)),
+                new ValidationConfig(MODE, Collections.singletonList(ValidationType.NON_EMPTY_STRING))
```

Comment:
mode is optional - https://razorpay.com/docs/partners/platform/onboard-businesses/integrate-oauth/integration-steps/#request-parameters

Reply:
Updated.